website/docs/cdktf/telemetry.html.md

```diff
@@ -3,9 +3,15 @@ layout: "docs"
 page_title: "Telemetry"
 sidebar_current: "cdktf"
 description: |-
-  TBD.
+  Product telemetry in CDK for Terraform
 ---
 
 # Telemetry
 
-TBD insert telemetry information.
\ No newline at end of file
+CDK for Terraform CLI interacts with a HashiCorp service called [Checkpoint](https://checkpoint.hashicorp.com) to report project metrics such as cdktf version, project language, provider name, platform name, and other details that help guide the project maintainers with feature and roadmap decisions. All HashiCorp projects, including Terraform Core, use Checkpoint. Read more about project metrics [here](https://github.com/hashicorp/terraform-cdk/issues/325).
+
+Starting with CDK for Terraform 0.6, this information includes a random UUID that uniquely identifies the machine, and new projects will have a project-specific UUID added to the cdktf.json file. The purpose of these UUIDs is to help team understand how the tool is used in order to help us prioritize features. The code that interacts with Checkpoint is part of CDK for Terraform CLI and can be read [here](https://github.com/hashicorp/terraform-cdk/blob/main/packages/cdktf-cli/lib/checkpoint.ts).
+
+When CDK for Terraform runs in a Continuous Integration tool, such as GitHub Actions, Jenkins, or GitLab, CDK for Terraform fingerprints the type of CI used, and includes that in telemetry instead of the UUID. The only information submitted is the type of CI system - no unique information about accounts, paths, workspaces, environment variables, or other potentially private information is shared. The purpose is to help the team plan where to focus future efforts, because running infrastructure as code tools in CI is a key workflow we seek to improve.
```

Comment:
```suggestion
CDK for Terraform fingerprints the type of CI used and includes that in telemetry instead of the UUID when CDK for Terraform runs in a Continuous Integration tool, such as GitHub Actions, Jenkins, or GitLab. The only information submitted is the type of CI system, and no unique information about accounts, paths, workspaces, environment variables, or other potentially private information is shared. This data helps the team plan where to focus future efforts, because running infrastructure as code tools in CI is a key workflow we seek to improve.
```